eml/screens/explore/Explore.jsx

```diff
@@ -1,87 +1,203 @@
 
-    const isFocused = useIsFocused()
-
-    const [selected, setSelected] = useState(-1);
-
-    const [nonActiveViews, setViews] = useState([]);
-    const [activeViews, setViews2] = useState([]);
-    // All, Sustainability, Finance, Other
-    const uniqueCategories = [{ key: -1, value: "Todos" }, { key: '635f9ae2991d8c6da796a1cc', value: "Sustentabilidade" }, { key: '6368be5d71e079ae8d537eb1', value: "Finança" }, { key: '639208a0f467689fde25b5fa', value: "De outros" }];
-
-    const [fontsLoaded] = useFonts({
-        VarelaRound_400Regular
-    })
-
-    async function loadViews() {
-        const courseList = await StorageService.getCourseList();
-        let componentPromises = courseList.map(({ title, iconPath, isActive, courseId, categoryId }, index) => {
-            if (isActive === false) {
-                if ((isActive && categoryId === selected) || (isActive && selected === -1)) {
-                    return <ActiveExploreCard key={index} title={title} courseId={courseId} iconPath={iconPath} />;
-                } else if ((!(isActive) && categoryId === selected) || (!(isActive) && selected === -1)) {
-                    return <ExploreCard key={index} title={title} courseId={courseId}></ExploreCard>
-                }
-            }
-        });
-        Promise.all(componentPromises).then(setViews);
-
-        let componentPromises2 = courseList.map(({ title, iconPath, isActive, courseId, categoryId }, index) => {
-            if (isActive === true) {
-                if ((isActive && categoryId === selected) || (isActive && selected === -1)) {
-                    return <ActiveExploreCard key={index} title={title} courseId={courseId} iconPath={iconPath} />;
-                } else if ((!(isActive) && categoryId === selected) || (!(isActive) && selected === -1)) {
-                    return <ExploreCard key={index} title={title} courseId={courseId}></ExploreCard>
-                }
-            }
-        });
-        Promise.all(componentPromises2).then(setViews2);
+  // Search text state
+  const [searchText, setSearchText] = useState('');
+  // Selected category state
+  const [selectedCategory, setSelectedCategory] = useState(null);
+
+  //Sets dummy data for courses (will be replaced with data from backend)
+  const [courses, setCourses] = useState([]);
+  const [subCourses, setSubCourses] = useState([]);
+  const [isSubscribed, setIsSubscribed] = useState([]);
+
+  const [refreshing, setRefreshing] = useState(false);
+  const navigation = useNavigation()
+
+
+  /**
+* Determines if the two arrays of courses are different and require an update.
+* @param {Array} courses1 - The first array of courses, typically representing the current state.
+* @param {Array} courses2 - The second array of courses, typically representing the new fetched data.
+* @returns {boolean} - Returns true if the two arrays are different and an update is required, otherwise false.
+*/
+  function shouldUpdate(courses1, courses2) {
```

Comment:
Why is this function defined again, when it is already defined in utilityFunctions.js